jetty-core/jetty-websocket/jetty-websocket-jetty-api/src/main/java/org/eclipse/jetty/websocket/api/Callback.java

```diff
@@ -64,18 +64,50 @@ default void succeed()
     }
 
     /**
-     * <p>
-     * Callback invoked when the write fails.
-     * </p>
+     * <p>Method to invoke to fail the callback.</p>
      *
-     * @param x the reason for the write failure
+     * @param x the failure
      */
     default void fail(Throwable x)
     {
     }
 
+    /**
+     * <p>Converts this callback into a {@link BiConsumer} that can be passed
+     * to {@link CompletableFuture#whenComplete(BiConsumer)}.</p>
+     * <p>When the {@link BiConsumer} is accepted, this callback is completed.</p>
+     *
+     * @return a {@link BiConsumer} that completes this callback
+     */
+    default BiConsumer<? super Void, ? super Throwable> asConsumer()
+    {
+        return (r, x) ->
+        {
+            if (x == null)
+                succeed();
+            else
+                fail(x);
+        };
+    }
```

Comment:
This is only used in a test, is it necessary?
If we expect applications to frequently use it, then we'd be better off making Callback extend `BiConsumer`

Reply:
@lorban and I discussed this, and it's more complicated than that.
We explored 3/4 solutions and liked none, so I think for now it's fine to keep it, as it simplifies a lot the code for users when dealing with `CompletableFuture` and `Callback`.